app/screens/apps_form/apps_form_field.tsx

```diff
@@ -0,0 +1,188 @@
+// Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
+// See LICENSE.txt for license information.
+
+import React, {useCallback} from 'react';
+import {KeyboardTypeOptions, View} from 'react-native';
+
+import AutocompleteSelector from '@components/autocomplete_selector';
+import Markdown from '@components/markdown';
+import BoolSetting from '@components/settings/bool_setting';
+import TextSetting from '@components/settings/text_setting';
+import {View as ViewConstants} from '@constants';
+import {AppFieldTypes} from '@constants/apps';
+import {useTheme} from '@context/theme';
+import {getMarkdownBlockStyles, getMarkdownTextStyles} from '@utils/markdown';
+import {makeStyleSheetFromTheme} from '@utils/theme';
+
+const TEXT_DEFAULT_MAX_LENGTH = 150;
+const TEXTAREA_DEFAULT_MAX_LENGTH = 3000;
+
+export type Props = {
+    field: AppField;
+    name: string;
+    errorText?: string;
+    value: AppFormValue;
+    onChange: (name: string, value: string | string[] | boolean) => void;
+    performLookup: (name: string, userInput: string) => Promise<AppSelectOption[]>;
+}
+
+const getStyleSheet = makeStyleSheetFromTheme((theme: Theme) => {
+    return {
+        markdownFieldContainer: {
+            marginTop: 15,
+            marginBottom: 10,
+            marginLeft: 15,
+        },
+        markdownFieldText: {
+            fontSize: 14,
+            color: theme.centerChannelColor,
+        },
+    };
+});
+
+function selectKeyboardType(subtype?: string): KeyboardTypeOptions {
+    switch (subtype) {
+        case 'email':
+            return 'email-address';
+        case 'number':
+            return 'numeric';
+        case 'tel':
+            return 'phone-pad';
+        case 'url':
+            return 'url';
+        default:
+            return 'default';
+    }
+}
+
```

Comment:
1/5 to create as a helper function and import. This is duplicated in `app/screens/interactive_dialog/dialog_element.tsx`